verify_test.go

```diff
@@ -42,33 +80,34 @@ func TestVerifyConcurrent(t *testing.T) {
 	for i := 0; i < 100; i++ {
 		wg.Add(1)
 		go func() {
-			skey, msg, err := Verify(smsg, kr)
+			defer wg.Done()
+			skey, msg, err := Verify(SingleVersionValidator(version), smsg, kr)
 			if err != nil {
 				t.Logf("input:      %x", in)
 				t.Logf("signed msg: %x", smsg)
 				t.Error(err)
+				return
```

Comment:
My bad, I skimmed over this and misread it as "putting this part in an inner function so that we can short circuit errors" rather than "adding a short-circuit to something that was already an inner function".